common/remote/sockfile.cpp

```diff
@@ -2315,6 +2315,21 @@ class CRemoteFileIO : implements IFileIO, public CInterface
         // then also send sMode, cFlags
         unsigned short sMode = parent->getShareMode();
         unsigned short cFlags = parent->getCreateFlags();
+        switch ((compatIFSHmode)_compatmode)
+        {
+            case compatIFSHnone:
+                sMode = IFSHnone;
+                break;
+            case compatIFSHread:
+                sMode = IFSHread;
+                break;
+            case compatIFSHwrite:
+                sMode = IFSHfull;
+                break;
+            case compatIFSHall:
+                sMode = IFSHfull;
+                break;
+        }
```

Comment:
It also feels like sMode that came from parent->getShareMode() is always going to be overwritten by this code, so I'm not sure I really understand the logic